FreeRTOS-Plus/Demo/AWS/Fleet_Provisioning_Windows_Simulator/Fleet_Provisioning_With_CSR_Demo/mqtt_operations.c

```diff
@@ -0,0 +1,1069 @@
+    xReturnStatus = prvGetNextFreeIndexForOutgoingPublishes( &ucPublishIndex );
+
+    if( xReturnStatus == false )
+    {
+        LogError( ( "Unable to find a free spot for outgoing PUBLISH message." ) );
+    }
+    else
+    {
+        LogDebug( ( "Published payload: %.*s",
+                    ( int ) xPayloadLength,
+                    ( const char * ) pcPayload ) );
+
+        /* This example publishes to only one topic and uses QOS1. */
+        pxOutgoingPublishPackets[ ucPublishIndex ].xPubInfo.qos = MQTTQoS1;
+        pxOutgoingPublishPackets[ ucPublishIndex ].xPubInfo.pTopicName = pcTopicFilter;
+        pxOutgoingPublishPackets[ ucPublishIndex ].xPubInfo.topicNameLength = usTopicFilterLength;
+        pxOutgoingPublishPackets[ ucPublishIndex ].xPubInfo.pPayload = pcPayload;
+        pxOutgoingPublishPackets[ ucPublishIndex ].xPubInfo.payloadLength = xPayloadLength;
+
+        /* Get a new packet id. */
+        pxOutgoingPublishPackets[ ucPublishIndex ].usPacketId = MQTT_GetPacketId( pxMqttContext );
+
+        /* Send PUBLISH packet. */
+        xMqttStatus = MQTT_Publish( pxMqttContext,
+                                    &pxOutgoingPublishPackets[ ucPublishIndex ].xPubInfo,
+                                    pxOutgoingPublishPackets[ ucPublishIndex ].usPacketId );
+
+        if( xMqttStatus != MQTTSuccess )
+        {
+            LogError( ( "Failed to send PUBLISH packet to broker with error = %s.",
+                        MQTT_Status_strerror( xMqttStatus ) ) );
+            prvCleanupOutgoingPublishAt( ucPublishIndex );
+            xReturnStatus = false;
+        }
+        else
+        {
+            LogDebug( ( "PUBLISH sent for topic %.*s to broker with packet ID %u.",
+                        usTopicFilterLength,
+                        pcTopicFilter,
+                        pxOutgoingPublishPackets[ ucPublishIndex ].usPacketId ) );
+        }
+    }
+
+    return xReturnStatus;
+}
+/*-----------------------------------------------------------*/
+
+bool xProcessLoop( void )
+{
+    bool xReturnStatus = false;
+    MQTTStatus_t xMqttStatus = MQTTSuccess;
+
+    xMqttStatus = MQTT_ProcessLoop( &xMqttContext, mqttopMQTT_PROCESS_LOOP_TIMEOUT_MS );
+
+    if( xMqttStatus != MQTTSuccess )
+    {
+        LogError( ( "MQTT_ProcessLoop returned with status = %s.",
+                    MQTT_Status_strerror( xMqttStatus ) ) );
+    }
+    else
+    {
+        LogDebug( ( "MQTT_ProcessLoop successful." ) );
+        xReturnStatus = true;
+    }
+
+    return xReturnStatus;
+}
+/*-----------------------------------------------------------*/
+
+static uint32_t prvGetTimeMs( void )
+{
+    TickType_t xTickCount = 0;
+    uint32_t ulTimeMs = 0UL;
+
+    /* Get the current tick count. */
+    xTickCount = xTaskGetTickCount();
+
+    /* Convert the ticks to milliseconds. */
+    ulTimeMs = ( uint32_t ) xTickCount * mqttopMILLISECONDS_PER_TICK;
+
+    /* Reduce ulGlobalEntryTimeMs from obtained time so as to always return the
```

Comment:
Is ulGlobalEntryTimeMs initialized anywhere?

Reply:
It seems that it isn't, and this is consistent across all of the AWS demos. From the other demos, it seems that normally it is initialized in the task defined in the "DemoTasks/*Example.c" file. I'll address this in the mqtt_demo_helpers PR to keep the AWS demos consistent.